cluster_pack/uploader.py

```diff
@@ -251,29 +251,53 @@ def _upload_env_from_venv(
 
     _logger.debug(f"Packaging current_packages={reqs}")
 
-    if force_upload or not _is_archive_up_to_date(package_path, reqs, resolved_fs):
-        _logger.info(
-            f"Zipping and uploading your env to {package_path}"
-        )
-
-        if include_editable:
-            editable_requirements = packaging.get_editable_requirements()
-        else:
-            editable_requirements = {}
+    if not force_upload and _is_archive_up_to_date(package_path, reqs, resolved_fs):
+        _logger.info(f"{package_path} already exists")
+        return
 
-        with tempfile.TemporaryDirectory() as tempdir:
-            archive_local = packer.pack(
-                output=f"{tempdir}/{packer.env_name()}.{packer.extension()}",
+    with tempfile.TemporaryDirectory() as tempdir:
+        env_copied_from_fallback_location = False
+        local_package_path = f'{tempdir}/{packer.env_name()}.{packer.extension()}'
+        local_fs, local_package_path = filesystem.resolve_filesystem_and_path(local_package_path)
+
+        fallback_path = os.environ.get('C_PACK_ENV_FALLBACK_PATH')
+        if not force_upload and fallback_path:
+            _logger.info(f"Copying env from {fallback_path} to {local_package_path}")
+            if fallback_path.startswith("http://") or fallback_path.startswith("https://"):
+                request.urlretrieve(fallback_path, local_package_path)
+            else:
+                fallback_fs, fallback_path = filesystem.resolve_filesystem_and_path(fallback_path)
+                fallback_fs.get(fallback_path, local_package_path)
+
+            _logger.info(f'Checking if requirements are met in {local_package_path}')
+            _dump_archive_metadata(local_package_path, reqs, local_fs)

```

Comment:
Sorry for the late reply but this actually doesn't work.
What you do is writing the metadata of you locally installed packages whereas what you need to do:
- get the list of packages from the pex file, 
- extract them or dump them as json
- compare them to reqs
- if the same upload the pex + metadatfile
- if not regenerate everything from scratch as locally you have a different set of installs

otherwise you will always use C_PACK_ENV_FALLBACK_PATH. it is not actually fallback.